### PR TITLE
DDF-2564: Updated cometd endpoint to properly startup

### PR DESCRIPTION
**Pull request by AzGoalie**
#### What does this PR do?
Fixes the cometd endpoint from failing to initialize on install and reboot.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@brendan-hofmann @coyotesqrl
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Build](https://github.com/orgs/codice/teams/build)
[Docs](https://github.com/orgs/codice/teams/docs)
[UI](https://github.com/orgs/codice/teams/ui)
[Security](https://github.com/orgs/codice/teams/security)
[Continuous Integration](https://github.com/orgs/codice/teams/continuous-integration)
[Solr](https://github.com/orgs/codice/teams/solr)
[IO](https://github.com/orgs/codice/teams/IO)
[OGC](https://github.com/orgs/codice/teams/OGC)
[Data](https://github.com/orgs/codice/teams/data)
[Core APIs](https://github.com/orgs/codice/teams/core-apis)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@pklinef
@stustison
##### On Windows
Extract DDF and install.
Make sure there are no warnings/errors in logs during install related to the search-endpoint or cometd.
After install verify that you can login to the standard search ui.
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java [hunk 1]
```diff
@@ -45,54 +43,39 @@
...
- * The CometdEndpoint binds the SearchService and the CometdServlet together. 
+ * The CometdEndpoint binds the SearchService and the CometdServlet together.
  * This is where the asynchronous endpoint is initially started.
  */
-public class CometdEndpoint {
+public class CometdEndpoint extends CometDServlet {
+    private static final long serialVersionUID = 1L;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(CometdEndpoint.class);
+    private static final transient Logger LOGGER = LoggerFactory.getLogger(CometdEndpoint.class);
```
Comment (coyotesqrl) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: static+transient makes no sense. Static fields aren't serialized.

Comment (AzGoalie) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: Woops, fixed.

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java [hunk 2]
```diff
@@ -45,54 +43,39 @@
...
  * This is where the asynchronous endpoint is initially started.
  */
-public class CometdEndpoint {
+public class CometdEndpoint extends CometDServlet {
+    private static final long serialVersionUID = 1L;
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CometdEndpoint.class);
 
-    private final CometDServlet cometdServlet;
+    private final transient FilterBuilder filterBuilder;
```
Comment (brendan-hofmann) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: Out of curiosity, why are any of these transient? I don't think the `CometdEndpoint` is serialized. Is intellij adding those?

Comment (coyotesqrl) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: It's extending `CometDServlet` now. Servlets are serializable.

Comment (brendan-hofmann) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: That makes sense. 👍

Comment (pklinef) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/endpoint/CometdEndpoint.java]: `CometDServlet` implents `Serializable` and `CometdEndpoint` now extends `CometDServlet`.

https://docs.cometd.org/current/apidocs/org/cometd/server/CometDServlet.html

**Pull request merged**